### PR TITLE
Prevents hitting Solr if we know there are no highlights for the user.

### LINKED ISSUE #1139: My Highlights should not query solr if no trophies are present
My Highlights does not show highlighted works.

The solr query being sent seems a bit off.  It includes "id:NEVER_USE_THIS_ID"

Full query:

Solr query: get select {"qt"=>"search", "facet.field"=>["resource_type_sim", "creator_sim", "tag_sim", "subject_sim", "language_sim", "based_near_sim", "publisher_sim", "file_format_sim"], "facet.query"=>[], "facet.pivot"=>[], "fq"=>["edit_access_group_ssim:public OR discover_access_group_ssim:public OR read_access_group_ssim:public OR edit_access_group_ssim:registered OR discover_access_group_ssim:registered OR read_access_group_ssim:registered OR edit_access_person_ssim:cam156@psu.edu OR discover_access_person_ssim:cam156@psu.edu OR read_access_person_ssim:cam156@psu.edu", "id:NEVER_USE_THIS_ID"], "hl.fl"=>[], "rows"=>10, "facet"=>true, "f.resource_type_sim.facet.limit"=>6, "f.creator_sim.facet.limit"=>6, "f.tag_sim.facet.limit"=>6, "f.subject_sim.facet.limit"=>6, "f.language_sim.facet.limit"=>6, "f.based_near_sim.facet.limit"=>6, "f.publisher_sim.facet.limit"=>6, "f.file_format_sim.facet.limit"=>6, "sort"=>"score desc, date_uploaded_dtsi desc"}

### PR DESCRIPTION
**Pull request by hectorcorrea**
Fixes #1139

### CODE REVIEW

File: app/controllers/concerns/sufia/my_controller_behavior.rb [hunk 1]
```diff
@@ -53,5 +53,35 @@ def index
         format.atom { render layout: false }
       end
     end
+
+    private
+
+      def query_solr
+        if params["controller"] == "my/highlights" && @user.trophy_works.count == 0
```
Comment (jcoyne) [inline app/controllers/concerns/sufia/my_controller_behavior.rb]: Can you just override `query_solr()` on the my/highlights controller?

Comment (hectorcorrea) [inline app/controllers/concerns/sufia/my_controller_behavior.rb]: Good idea.

**Comment by grosscol** [comment]
Could you add a spec that ensures we won't call solr in the future when trophies count is zero?

**Comment by hectorcorrea** [comment]
@grosscol Test has been added.

**Pull request merged**